### PR TITLE
[2.0.3][dolphinscheduler-server] process instance is always running 

### PR DESCRIPTION
**Pull request by zwZjut**
## Purpose of the pull request

https://github.com/apache/dolphinscheduler/issues/7700
https://github.com/apache/dolphinscheduler/issues/7698

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecService.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                    startProcessFailedMap.putIfAbsent(processInstanceId, workflowExecuteThread);
+                } else {
+                    startProcessFailedMap.remove(processInstanceId);
+                }
+                filterMap.remove(processInstanceId);
+            }
+
+            @Override
+            public void onFailure(Throwable throwable) {
+                logger.error("handle events {} failed.", processInstanceId);
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecService.java]: Is it better to merge log like:
```suggestion
                logger.error("handle events {} failed", processInstanceId, throwable);
```

**Review by caishunfeng — COMMENTED** [review]
LGTM

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/StateWheelExecuteThread.java [hunk 2]
```diff
@@ -176,4 +192,12 @@ private void addEvent(StateEvent stateEvent) {
         workflowExecuteThread.addStateEvent(stateEvent);
     }
 
+    private void check4StartProcessFailed() {
+        if (startProcessFailedMap.isEmpty()) {
+            return;
+        }
+        for (WorkflowExecuteThread workflowExecuteThread : this.startProcessFailedMap.values()) {
+            masterExecService.execute(workflowExecuteThread);
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/StateWheelExecuteThread.java:200]: This thread may have concurrency issues with other threads.

**Comment by lenboo** [comment]
@zwZjut @caishunfeng  Well done! At the same time , i think we can add some descriptions what these changes would affect. 
Eg : 
1.  The killing process of workflow.
2. ...

**Review by lenboo — APPROVED** [review]
+1

**Comment by caishunfeng** [comment]
Mainly stability optimization:

- Fixed that in some cases concurrent consumption of events resulted in no consumption, resulting in the workflow instance running all the time;
- Optimization of kill logic;
- Workflow instance start exception retry;

**Comment by caishunfeng** [comment]
Mainly stability optimization:
Fixed that in some cases concurrent consumption of events resulted in no consumption, resulting in the workflow instance running all the time;
Optimization of kill logic;
Workflow instance start exception retry;

**Pull request merged**